stateproof/builder.go

```diff
@@ -327,19 +329,45 @@ func (spw *Worker) broadcastSigs(brnd basics.Round, proto config.ConsensusParams
 	}
 }
 
-func (spw *Worker) deleteOldSigs(nextStateProof basics.Round) {
+func lowestRoundToRemove(currentHdr bookkeeping.BlockHeader) basics.Round {
+	proto := config.Consensus[currentHdr.CurrentProtocol]
+	nextStateProofRnd := currentHdr.StateProofTracking[protocol.StateProofBasic].StateProofNextRound
+	if proto.StateProofInterval == 0 {
+		return nextStateProofRnd
+	}
+
+	recentRoundOnRecoveryPeriod := basics.Round(uint64(currentHdr.Round) - uint64(currentHdr.Round)%proto.StateProofInterval)
+	oldestRoundOnRecoveryPeriod := recentRoundOnRecoveryPeriod.SubSaturate(basics.Round(proto.StateProofInterval * proto.StateProofRecoveryInterval))
```

Comment:
Here, shouldn't the protocol used be the one from round `recentRoundOnRecoveryPeriod` instead of `proto`, which is the round of `currentHdr`?

Reply:
from my understanding, 
in the current implementation, if we choose to change the `StateProofRecoveryInterval` it will take effect immediately. if we change the code and take  `StateProofRecoveryInterval` from `recentRoundOnRecoveryPeriod` , changing `StateProofRecoveryInterval` might take effect one interval after.

Do you think I got it right?
if so, would it be simpler to keep it as is and a change `StateProofRecoveryInterval` will take effect immediately?